sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/HCatalogIOTest.java

```diff
@@ -0,0 +1,342 @@
+import org.junit.rules.ExpectedException;
+
+
+/**
+ *Test for HCatalogIO.
+ */
+public class HCatalogIOTest extends HCatBaseTest implements Serializable {
+
+  @Rule
+  public final transient TestPipeline pipeline = TestPipeline.create();
+
+  @Rule
+  public transient ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  /**
+   * Perform end-to-end test of Read operation.
+   */
+  public void testReadE2ESuccess() throws CommandNeedRetryException,
+  IOException, ClassNotFoundException {
+
+    Map<String, String> map = insertTestData();
+
+    PCollection<String> output = pipeline.apply(HCatalogIO.read()
+        .withConfigProperties(map)
+        .withTable(HCatalogIOTestUtils.TEST_TABLE_NAME))
+        .apply(ParDo.<DefaultHCatRecord, String>of(new DoFn<DefaultHCatRecord, String>() {
+          @ProcessElement
+          public void processElement(ProcessContext c) {
+            c.output(c.element().get(0).toString());
+          }
+        }));
+    PAssert.thatSingleton(output.apply("Count All", Count.<String>globally()))
+    .isEqualTo(100L);
+    pipeline.run();
+  }
+
+  @Test
+  /**
+   * Perform end-to-end test of Write operation.
+   */
+  public void testWriteE2ESuccess() throws CommandNeedRetryException,
+  IOException, ClassNotFoundException {
+    createTestTable();
+    pipeline.apply(
+        Create.of(
+            toHCatRecord(0), toHCatRecord(1), toHCatRecord(2), toHCatRecord(3),
+            toHCatRecord(4), toHCatRecord(5), toHCatRecord(6), toHCatRecord(7),
+            toHCatRecord(8), toHCatRecord(9), toHCatRecord(10), toHCatRecord(11),
+            toHCatRecord(12), toHCatRecord(13), toHCatRecord(14), toHCatRecord(15),
+            toHCatRecord(16), toHCatRecord(17), toHCatRecord(18), toHCatRecord(19),
+            toHCatRecord(20), toHCatRecord(21), toHCatRecord(22), toHCatRecord(23),
+            toHCatRecord(24), toHCatRecord(25), toHCatRecord(26), toHCatRecord(27),
+            toHCatRecord(28), toHCatRecord(29), toHCatRecord(30), toHCatRecord(31),
+            toHCatRecord(32), toHCatRecord(33), toHCatRecord(34), toHCatRecord(35),
+            toHCatRecord(36), toHCatRecord(37), toHCatRecord(38), toHCatRecord(39))
+        )
+        .apply(HCatalogIO.write()
+            .withConfigProperties(getConfigProperties())
+            .withTable(TEST_TABLE_NAME));
+    pipeline.run();
+    Assert.assertTrue(HCatalogIOTestUtils.getRecordsCount(getConfigProperties()) == 40);
+  }
+
+  /**
+   * Test of Write to a non-existent table.
+   */
+  @Test
+  public void testWriteFailureTableDoesNotExist() throws CommandNeedRetryException,
+  IOException, ClassNotFoundException {
+    thrown.expectCause(isA(UserCodeException.class));
+    thrown.expectMessage(containsString("org.apache.hive.hcatalog.common.HCatException"));
+    thrown.expectMessage(containsString("NoSuchObjectException"));
+    pipeline.apply(
+        Create.of(
+            toHCatRecord(0), toHCatRecord(1), toHCatRecord(2), toHCatRecord(3),
```

Comment:
Same. Moreover, in this test the Create.of() data doesn't matter at all, because the test verifies failure to write it.